src/helm/proxy/clients/critique_client.py

```diff
@@ -0,0 +1,208 @@
+            #
+            # Example cache value:
+            # {"id": "17e323f1-f7e4-427c-a2d5-456743aba8"}
+            #
+            # Note:
+            # We do not cache the additional fields returned by surge.Project.create()
+            # because they are mutable server-side, and server-side mutations may cause
+            # stale cache issues.
+            task_response, is_cached = self._cache.get({"project_id": project_id, "fields": fields}, create_surge_task)
+        task_id = task_response["id"]
+        if is_cached:
+            hlog(f"Reusing existing Surge AI task: {task_id}")
+        else:
+            hlog(f"Creating new Surge AI task: {task_id}")
+        return task_id
+
+    def _get_worker_responses(self, task_id: str, questions: List[CritiqueQuestionTemplate]) -> List[CritiqueResponse]:
+        task = surge.Task.retrieve(task_id)
+        return [
+            CritiqueResponse(
+                id=task_response.id,
+                respondant_id=task_response.worker_id,
+                answers={question.name: task_response.data[question.name] for question in questions},
+            )
+            for task_response in task.responses
+        ]
+
+    def make_critique_request(self, request: CritiqueRequest) -> Optional[CritiqueRequestResult]:
+        """Create a task on Surge AI and fetch responses from Surge AI if available.
+
+        Returns CritiqueRequestResult if worker answers are complete, or None otherwise.
+        The intended use is to call it once to create the task, wait a while, and then call it
+        later to fetch answers.
+
+        First, attempt to find a Surge AI project for the template. If one exists, reuse that project.
+        Otherwise, create a new project using the template.
+
+        Second, attempt to find a Surge AI task inside this project for the fields. If one exists,
+        reuse that task. Otherwise, create a new task inside the project using the fields.
+
+        Finally, check if responses are available by checking if the number of workers who have responded
+        is equal to the requested number of workers. If so, return those responses.
+
+        This method is idempotent, because projects and tasks are not created if they already exist.
+
+        The cache will store the mappings from template to Surge AI Project ID and from fields to Surge AI
+        question ID. If the cache is deleted, the mappings will be lost, and this method will not be able
+        to fetch results from the previous projects and tasks, and will have to create new projects and tasks.
+        Note that worker responses are currently not cached."""
+        project_id = self._get_or_create_surge_project(request.template)
+        task_id = self._get_or_create_task(project_id, request.fields)
+        worker_responses = self._get_worker_responses(task_id, request.template.questions)
+        if len(worker_responses) < request.template.num_respondants:
+            return None
```

Comment:
Can we return partial results and have a flag that says they are partial?  It would be nice to compute stats or show the results even if we're not done yet.

Reply:
I added partial result returning. Not sure if we need a flag because users can just check if `len(result.responses) == request.template.num_respondents`